Error indexing some arks causing traject to crash and not finish the file

`2021-07-28T11:54:20-04:00 [ERROR] 4241 : Unexpected error on record id `994858723506421/994858723506421` at file position 15690`
11:57
2021-07-28T11:48:08-04:00 [ERROR] 23980 : Unexpected error on record id `993588923506421/993588923506421` at file position 71377

Check datadog:

* [994858723506421](https://app.datadoghq.com/logs?event&from_ts=1627326807126&index=&live=true&query=service%3Atraject++ERROR+994858723506421&to_ts=1627499607126)
* [993588923506421](https://app.datadoghq.com/logs?query=service%3Atraject%20%20%20ERROR%20993588923506421&index=&from_ts=1627326807126&to_ts=1627499607126&live=true)

MMS_ID `993588923506421` is found on file https://bibdata-alma.princeton.edu/dump_files/1173 (https://bibdata-alma.princeton.edu/dumps/21.json)

MMS_ID `994858723506421` is found on file https://bibdata-alma.princeton.edu/dump_files/1185 (https://bibdata-alma.princeton.edu/dumps/21.json)

--- a/marc_to_solr/lib/uri_ark.rb
+++ b/marc_to_solr/lib/uri_ark.rb
@@ -41,7 +41,7 @@ class URI::ARK < URI::Generic
     # Extract the components from the ARK URL into member variables
     def extract_components!
       raise StandardError, "Invalid ARK URL using: #{self.to_s}" unless self.class.princeton_ark?(url: self)
-      m = /\:\/\/(.+)\/ark\:\/(.+)\/(.+)\/?/.match(self.to_s)
+      m = /\/(.+)\/ark\:\/(.+)\/(.+)\/?/.match(self.to_s)
 
       @nmah = m[1]
       @naan = m[2]

--- a/spec/marc_to_solr/lib/uri_ark_spec.rb
+++ b/spec/marc_to_solr/lib/uri_ark_spec.rb
@@ -5,6 +5,7 @@ RSpec.describe URI::ARK do
     let(:princeton_ark) { 'http://arks.princeton.edu/ark:/88435/9k41zd556' }
     let(:invalid_princeton_ark) { 'http:http://arks.princeton.edu/ark:/88435/9k41zd556' }
     let(:non_princeton_ark) { 'https://nls.ldls.org.uk/welcome.html?ark:/81055/vdc_100052020059.0x000001' }
+    let(:single_slash_ark) { 'http:/arks.princeton.edu/ark:/88435/ff365d62r/pdf' }
     it 'Princeton arks return true' do
       expect(described_class.princeton_ark?(url: princeton_ark)).to eq true
     end
@@ -13,6 +14,10 @@ RSpec.describe URI::ARK do
     end
     it 'arks that include http twice the url return false ' do
       expect(described_class.princeton_ark?(url: invalid_princeton_ark)).to eq false
+    end
+    it 'handles single forward slash gracefully' do
+      ark = described_class.parse(url: single_slash_ark)
+      expect(ark.to_s).to eq "http::80/arks.princeton.edu/ark:/88435/ff365d62r/pdf"
     end
   end
 end

